addons/hr_recruitment/models/hr_job.py

```diff
@@ -141,17 +154,21 @@ def _compute_extended_interviewer_ids(self):
 
     def _compute_is_favorite(self):
         for job in self:
-            job.is_favorite = self.env.user in job.favorite_user_ids
+            job.is_favorite = self.env.user.employee_id in job.favorite_recruiters
 
     def _inverse_is_favorite(self):
+        employee = self.env.user.employee_id
+        if not employee:
+            return
+
         unfavorited_jobs = favorited_jobs = self.env['hr.job']
         for job in self:
-            if self.env.user in job.favorite_user_ids:
+            if employee in job.favorite_recruiters:
                 unfavorited_jobs |= job
             else:
                 favorited_jobs |= job
-        favorited_jobs.write({'favorite_user_ids': [(4, self.env.uid)]})
-        unfavorited_jobs.write({'favorite_user_ids': [(3, self.env.uid)]})
+        favorited_jobs.write({'favorite_recruiters': [(4, employee.id)]})
+        unfavorited_jobs.write({'favorite_recruiters': [(3, employee.id)]})
```

Comment:
Here you also have the option to use the "new" `Command` commands instead of the magic numbers, but it's up to you what you prefer.
```suggestion
        favorited_jobs.write({'favorite_recruiters': [Command.link(employee.id)]})
        unfavorited_jobs.write({'favorite_recruiters': [Command.unlink(employee.id)]})
```